src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,160 @@
+package duke;
+
+import java.time.format.DateTimeParseException;
+import java.util.Scanner;
+
+/**
+ * Parser is a class that deals with decoding user inputs
+ */
+public class Parser {
+    private TaskList taskList;
+    private Scanner scanner;
+    private boolean isScanning;
+
+
+    /**
+     * Constructor for Parser
+     *
+     * @param taskList The taskList that may be pre-populated from a previous save.
+     */
+    public Parser(TaskList taskList) {
+        this.taskList = taskList;
+        this.scanner = new Scanner(System.in);
+        this.isScanning = true;
+    }
+
+    public boolean isScanning() {
+        return this.isScanning;
+    }
+
+    /**
+     * Takes in user input, then updates the taskList with the appropriate Tasks
+     *
+     * @throws DukeException When command is unclear or string is given in a wrong format
+     */
+    public void parse() throws DukeException {
```

Comment:
Can this be replaced with switch statements? Switch statements can reduce the number of times doing repeated checks such as (line.length <= 7) which will boost the readability of the program.